lock-api/src/test/java/com/palantir/lock/watch/ConjureLockRequestMetadataUtilsTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.timelock.api.ConjureChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureCreatedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureDeletedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureLockDescriptorListChecksum;
+import com.palantir.atlasdb.timelock.api.ConjureLockRequestMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureUnchangedChangeMetadata;
+import com.palantir.atlasdb.timelock.api.ConjureUpdatedChangeMetadata;
+import com.palantir.common.streams.KeyedStream;
+import com.palantir.conjure.java.lib.Bytes;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.StringLockDescriptor;
+import com.palantir.lock.watch.ConjureLockRequestMetadataUtils.ConjureMetadataConversionResult;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.util.IndexEncodingUtils;
+import com.palantir.util.IndexEncodingUtils.KeyListChecksum;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class ConjureLockRequestMetadataUtilsTest {
+    private static final LockDescriptor LOCK_1 = StringLockDescriptor.of("lock1");
+    private static final LockDescriptor LOCK_2 = StringLockDescriptor.of("lock2");
+    private static final LockDescriptor LOCK_3 = StringLockDescriptor.of("lock3");
+    private static final LockDescriptor LOCK_4 = StringLockDescriptor.of("lock4");
+    private static final List<LockDescriptor> LOCK_LIST = ImmutableList.of(LOCK_1, LOCK_2, LOCK_3, LOCK_4);
+    // LinkedHashSet remembers insertion order, which is important for the tests below.
+    private static final Set<LockDescriptor> LOCK_SET = new LinkedHashSet<>(LOCK_LIST);
```

Comment:
I think we can just inline this below, seeing as it's only used in one test. Unless you can see this being used in the future? Only other thing that sketches me out is having a mutable `private static final`

Reply:
ImmutableSet also remembers insertion order, so there is actual no need to use the mutable `LinkedHashSet`